pkg/build/builder/common.go

```diff
@@ -82,3 +92,124 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds with the same name in the same namespace.
```

Comment:
@smarterclayton @bparees are you fine with the way we generate the temporary random build tag?

If you have a build named "ruby-hello-world-1" in the "sample-app" namespace, the build tag will look like: `sample-app/ruby-hello-world-1:f1f85ff5`, as proposed by @smarterclayton.

Only in the face of very long (~255 chars) namespace + name, we'll have a less human friendly name based on the SHA1 hash of namespace + name. We opted for that instead of truncation or other algorithms because we'd have to account for producing invalid names like "longnamespace/:f1f85ff5". Using a hash for the repo part guarantees that we produce a valid name with a known length. Again, this is a corner case.